### ISSUE TITLE
Incorrect code parsing

### ISSUE THREAD

**Issue by kukuruku20** [VirusTotal/yara-x#136]
As a result of how `pest` library is implemented and usage of `WHITESPACE` rule in the grammar which silently consumes all white spaces,
it is possible to correctly parse wrong source code e.g.:
```
privateruleexample { condition: true }
ruletest{condition:example}
``` 
Which results in following AST:
```
[
    Rule {
        flags: RuleFlags {
            mask: 1,
        },
        identifier: Ident {
            span: Span {
                source_id: SourceId(
                    0,
                ),
                start: 11,
                end: 18,
            },
            name: "example",
        },
        tags: None,
        meta: None,
        patterns: None,
        condition: True {
            span: Span {
                source_id: SourceId(
                    0,
                ),
                start: 32,
                end: 36,
            },
        },
    },
    Rule {
        flags: RuleFlags {
            mask: 0,
        },
        identifier: Ident {
            span: Span {
                source_id: SourceId(
                    0,
                ),
                start: 42,
                end: 46,
            },
            name: "test",
        },
        tags: None,
        meta: None,
        patterns: None,
        condition: Ident(
            Ident {
                span: Span {
                    source_id: SourceId(
                        0,
                    ),
                    start: 57,
                    end: 64,
                },
                name: "example",
            },
        ),
    },
]
```

**Comment by plusvic** [comment]
I'm working on a new version of the parser that fixes this problem.

**Comment by plusvic** [comment]
Fixed in the 0.6.0 release.
